### PR TITLE
feat: add node prefix import for Deno support

### PR DESCRIPTION
**Pull request by benmccann**
tests don't pass because of https://github.com/standard-things/esm/issues/904

### CODE REVIEW

**Comment by lukeed** [comment]
~~Sorry, can you remove the eslint stuff~~

nvm i could

File: package.json [hunk 1]
```diff
@@ -7,11 +7,14 @@
...
   "scripts": {
     "build": "bun run scripts/build.ts",
+    "lint": "eslint",
     "test": "uvu -r esm -i util -i bench packages test"
   },
   "devDependencies": {
     "bump": "1.0.0-next.1",
     "bundt": "1.1.2",
+    "eslint": "^9.9.1",
+    "eslint-plugin-n": "^17.10.2",
```
Comment (lukeed) [inline package.json]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -7,11 +7,14 @@
   },
   "scripts": {
     "build": "bun run scripts/build.ts",
+    "lint": "eslint",
```
Comment (lukeed) [inline package.json]:
```suggestion
```

**Review by lukeed — APPROVED** [review]
Merging despite the failing tests.

Future releases will continue to include the `node:` prefixes for `import` conditions, but omit the prefixes for `require` (CommonJS) conditions. This will maximize backwards Node compat, since [below 14](https://nodejs.org/api/esm.html#node-imports) will more than likely not understand `exports` maps and not find the `import` entries anyway

**Comment by xqin** [comment]
It is OK to add the node: prefix, but please update the major version number to 2.x.

The current release is a breaking change for existing applications.

**Comment by lukeed** [comment]
The `next` track of Polka is "next" -- it's not stable release yet. In fact, `next` is targeted **to be** the 1.0 release.
The master / stable branch is currently `0.5.x`

This is also why you install polka via `polka@next` -- you're opting into the next release channel.

Apologies about the breaking change, but working on finding a working solution for modern **and** more legacy codebases. It'd help if you provided information in the issue, not here.

**Comment by xqin** [comment]
> The `next` track of Polka is "next" -- it's not stable release yet. In fact, `next` is targeted **to be** the 1.0 release. The master / stable branch is currently `0.5.x`
> 
> This is also why you install polka via `polka@next` -- you're opting into the next release channel.
> 
> Apologies about the breaking change, but working on finding a working solution for modern **and** more legacy codebases. It'd help if you provided information in the issue, not here.


Dependency:

```
webpack-bundle-analyzer@4.10.2  -> sirv@^2.0.3 -> @polka/url^1.0.0-next.24
```

The question is: why does sirv depend on the next version of @polka/url, but the stable version?

And it seems that the sirv package is also maintained by you.

**Comment by lukeed** [comment]
> It'd help if you provided information in the issue, **not here.**

**Pull request merged**